test/functional/connection.ts

```diff
@@ -240,6 +240,21 @@ describe("connection", function () {
         done();
       });
     });
+
+    it("should reject on initial connection error", function (done) {
+      const stub = sinon
+        .stub(net.Socket.prototype, "connect")
+        .callsFake((options) => {
+          net.Socket.prototype.emit("error");
```

Comment:
this need to be async, .connect() doesnt throw in the same tick - wrap this in process.nextTick

https://nodejs.org/api/net.html#net_socket_connect

Reply:
Thanks, it seems to be capturing the behaviour I intended it to now